### PR TITLE
Add login and logout views.

### PR DESCRIPTION
**Pull request by sylmak**
Add login and logout views.

Also fix format command in package.json to format .vue files.

### CODE REVIEW

File: src/components/LogOut.vue [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { router } from '../router';
+
+const auth = getAuth();
+
+const props = defineProps(REDIRECT_URI_PROP);
+
+function logOut() {
+  signOut(auth).then(() => {
+    if (props.redirectUri) {
+      window.location.href = 'http://' + props.redirectUri;
```
Comment (skeshive) [inline src/components/LogOut.vue]: The redirectUri should already contain https. Please confirm with Jesus though

File: src/components/LogOut.vue [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import { onMounted } from 'vue';
+
+import { REDIRECT_URI_PROP } from '../common';
+import { router } from '../router';
+
+const auth = getAuth();
+
+const props = defineProps(REDIRECT_URI_PROP);
+
+function logOut() {
```
Comment (skeshive) [inline src/components/LogOut.vue:12]: Should we also clear the authentication cookies or will the Apigee logic be added in a follow up CL?

File: src/components/LogIn.vue [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+async function signIn() {
+  try {
+    await signInWithEmailAndPassword(auth, email.value, password.value);
+  } catch (error) {
+    // TODO: Clean up error message.
+    errorMessage.value = (error as Error).message;
+    return;
+  }
+  // Clear the error if this sign in attempt does not throw.
```
Comment (skeshive) [inline src/components/LogIn.vue]: does not throw an error?

File: src/components/LogOut.vue (hunk 1 above)
Comment (skeshive) [reply src/components/LogOut.vue]: same comment for login view

File: src/components/LogIn.vue [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    <div class="error-message">{{ errorMessage }}</div>
+    <div class="email">
+      <label for="email">Email</label>
+      <input type="email" v-model="email" />
+    </div>
+
+    <div class="password">
+      <label for="password">Password</label>
+      <input type="password" v-model="password" />
+    </div>
```
Comment (skeshive) [inline src/components/LogIn.vue:59]: We're missing a reset password option here

File: src/components/LogIn.vue [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+// Fields
+const email = ref();
+const password = ref();
+const errorMessage = ref();
+
+async function signIn() {
+  try {
+    await signInWithEmailAndPassword(auth, email.value, password.value);
+  } catch (error) {
+    // TODO: Clean up error message.
```
Comment (skeshive) [inline src/components/LogIn.vue:25]: Can we display this error in the UI? Mocks show two different errors in the UI: https://www.figma.com/design/g2iKY4XSSt2HnfCTRH9DZB/Climate-IQ-Website?node-id=2203-2577

File: src/components/LogOut.vue (hunk 2 above)
Comment (sylmak) [reply src/components/LogOut.vue:12]: Apigee logic will be added in a follow up CL. No cookies are currently set.

File: src/components/LogIn.vue (hunk 5 above)
Comment (sylmak) [reply src/components/LogIn.vue:25]: I'm looking at the mocks and I'm not sure which type of error should be shown where. Currently, this code just shows any error at the very top of the form.

Comment (skeshive) [reply src/components/LogIn.vue:25]: Hmm can we show both errors at the top for now? Incorrect username/password or account doesn't exist seem to be the two errors in the mocks

**Review by skeshive — APPROVED** [review]

File: src/components/LogIn.vue (hunk 5 above)
Comment (skeshive) [reply src/components/LogIn.vue:25]: Is http://35.230.74.0/auth/ up to date? When I tried creating a new account with an existing email, nothing in the UI indicates an error but I see an error (email already exists) in the console

Comment (sylmak) [reply src/components/LogIn.vue:25]: The pushed version isn't up to date.

**Pull request merged**